packages/data/src/batcher.ts

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ReadStream } from 'fs';
+import { Connection, Logger, Messages, SfdxError } from '@salesforce/core';
+import parse = require('csv-parse');
+import { Batch, BatchInfo, BatchResultInfo, JobInfo } from 'jsforce';
+import { UX } from '@salesforce/command';
+import { Job as jsforceJob } from 'jsforce/job';
+
+const BATCH_RECORDS_LIMIT = 10000;
+const POLL_FREQUENCY_MS = 5000;
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/data', 'batcher');
+
+export type Batches = Array<Array<Record<string, string>>>;
+export type Job = jsforceJob & { id: string };
+
+export type BulkResult = {
+  $: {
+    xmlns: string;
+  };
+  id: string;
+  jobId: string;
+  state: string;
+  createdDate: string;
+  systemModStamp: string;
+  numberRecordsProcessed: string;
+  numberRecordsFailed: string;
+  totalProcessingTime: string;
+  apiActiveProcessingTime: string;
+  apexProcessingTime: string;
+};
+
+export class Batcher {
```

Comment:
As we put stuff into this data library we need to be thinking about how people can use it for their own purposes. As it is now, I think it's a little unclear how people would use this or why they would use it.

So I'm wondering if there's a way to make this a little more user friendly. My first inclination is to rename it to `Job` and have it include all the logic for creating the job that's currently in the plugin. That way the commands merely wrap around this class.

Let me know if that doesn't make sense